bin/facades.php

```diff
@@ -185,70 +192,221 @@ function resolveDocParamType($method, $parameter)
+ * Handle conditional types.
  *
- * Unfortunately the @template tag is not currently working with the @method
- * docblocks, so we are stripping them out.
+ * @param  \ReflectionMethodDecorator  $method
+ * @param  \PHPStan\PhpDocParser\Ast\Type\ConditionalTypeNode  $typeNode
+ * @return string
+ */
+function handleConditionalType($method, $typeNode)
+{
+    if (
+        in_array($method->getname(), ['pull', 'get']) &&
+        $method->getDeclaringClass()->getName() === Repository::class
+    ) {
+        return 'mixed';
+    }
+
+    echo 'Found unknown conditional type. You will need to update the `handleConditionalType` to handle this new conditional type.';
+    exit(1);
+}
+
+/**
+ * Handle unknown identifier types.
  *
  * @param  \ReflectionMethodDecorator  $method
- * @param  string  $types
+ * @param  \PHPStan\PhpDocParser\Ast\Type\IdentifierTypeNode  $typeNode
  * @return string
  */
-function stripGenerics($method, $types)
+function handleUnknownIdentifierType($method, $typeNode)
 {
     if (
-        'enum' === $method->getName() &&
-        Request::class === $method->getDeclaringClass()->getName()
+        $typeNode->name === 'TCacheValue' &&
+        $method->getDeclaringClass()->getName() === Repository::class
     ) {
-        return Str::replace('TEnum', 'object', $types);
+        return 'mixed';
     }
 
     if (
-        'when' === $method->getName() &&
+        $typeNode->name === 'TWhenParameter' &&
         in_array(Conditionable::class, class_uses_recursive($method->getDeclaringClass()->getName()))
     ) {
-        return Str::of($types)
-            ->replace(['TWhenParameter', 'TWhenReturnType'], 'mixed')
-            ->replace('mixed|null', 'mixed')
-            ->toString();
+        return 'mixed';
     }
 
     if (
-        'unless' === $method->getName() &&
+        $typeNode->name === 'TWhenReturnType' &&
         in_array(Conditionable::class, class_uses_recursive($method->getDeclaringClass()->getName()))
     ) {
-        return Str::of($types)
-            ->replace(['TUnlessParameter', 'TUnlessReturnType'], 'mixed')
-            ->replace('mixed|null', 'mixed')
-            ->toString();
+        return 'mixed';
+    }
+
+    if (
+        $typeNode->name === 'TUnlessParameter' &&
+        in_array(Conditionable::class, class_uses_recursive($method->getDeclaringClass()->getName()))
+    ) {
+        return 'mixed';
     }
 
-    return $types;
+    if (
+        $typeNode->name === 'TUnlessReturnType' &&
+        in_array(Conditionable::class, class_uses_recursive($method->getDeclaringClass()->getName()))
+    ) {
+        return 'mixed';
+    }
+
+    if (
+        $typeNode->name === 'TEnum' &&
+        $method->getDeclaringClass()->getName() === Request::class
+    ) {
+        return 'object';
+    }
+
+    echo 'Found unknown type: '.$typeNode->name;
+    echo PHP_EOL;
+    echo 'You will need to update the `handleUnknownIdentifierType` to handle this new type / generic.';
+    exit(1);
```

Comment:
This will now error when encountering unknown generics / classes, which means we won't auto-commit wrong values.